docs/getting_started/filtering.md

```diff
@@ -0,0 +1,108 @@
+```
+
+```python
+print(tree.pretty())
+```
+
+```shell
+filter
+  expression
+    expression_clause
+      expression_phrase
+        comparison
+          property_first_comparison
+            property	nelements
+            value_op_rhs
+              <
+              value
+                number	3
+```
+
+```python
+tree = p.parse('_mp_bandgap > 5.0 AND _cod_molecular_weight < 350')
+print(tree.pretty())
+```
+
+```shell
+filter
+  expression
+    expression_clause
+      expression_phrase
+        comparison
+          property_first_comparison
+            property	_mp_bandgap
+            value_op_rhs
+              >
+              value
+                number	5.0
+      expression_phrase
+        comparison
+          property_first_comparison
+            property	_cod_molecular_weight
+            value_op_rhs
+              <
+              value
+                number	350
+```
+
+## Flow for parsing a user-supplied filter and converting to a backend query
+
+`optimade.filterparser.LarkParser` will take user input to generate a `lark.Tree` and feed that to a `lark.Transformer`.
+For example, `optimade.filtertransformers.mongo.MongoTransformer` will turn the tree into something useful for a MongoDB backend:
+
+```python
+# Example: Converting to MongoDB Query Syntax
+from optimade.filtertransformers.mongo import MongoTransformer
+
+transformer = MongoTransformer()
+
+tree = p.parse('_mp_bandgap > 5.0 AND _cod_molecular_weight < 350')
+query = transformer.transform(tree)
+print(query)
+```
+
+```json
+{
+    "$and": [
+        {"_mp_bandgap": {"$gt": 5.0}},
+        {"_cod_molecular_weight": {"$lt": 350.0}}
+    ]
+}
+```
+
+
+## Developing new filter transformers
+
+In order to support a new backend, you will need to create a new filter transformer that inherits from the [`BaseTransformer`][optimade.filtertransformers.base_transformer.BaseTransformer].
+This transformer will need to override the methods that match the particular grammatical constructs in the Lark grammar in order to construct a query.
+Two examples can be found for MongoDB ([`MongoTransformer`][optimade.filtertransformers.mongo.MongoTransformer]) and Elasticsearch ([`ElasticTransformer`][optimade.filtertransformers.elasticsearch.ElasticTransformer]).
+
+In some cases, you may also need to write a new [`EntryCollection`][optimade.server.entry_collections.entry_collections.EntryCollection] that receives the transformed filter as an argument to its private `._run_db_query()` method.
+This class handles the connections to the underlying database, formatting of the response in an OPTIMADE format, and other API features such as sorting and pagination.
+Again, the examples for MongoDB ([`MongoCollection`][optimade.server.entry_collections.mongo.MongoCollection]) and Elasticsearch ([`ElasticCollection`][optimade.server.entry_collections.elasticsearch.ElasticCollection]) should be helpful.
```

Comment:
```suggestion
Again, this class can be a sub-class of [`EntryCollection`][optimade.server.entry_collections.entry_collections.EntryCollection], where the examples for MongoDB ([`MongoCollection`][optimade.server.entry_collections.mongo.MongoCollection]) and Elasticsearch ([`ElasticCollection`][optimade.server.entry_collections.elasticsearch.ElasticCollection]) should be helpful.
```

Reply:
Reworded the first sentence in this paragraph to the same effect